x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/InitializePolicyContextStep.java

```diff
@@ -34,8 +37,28 @@ public ClusterState performAction(Index index, ClusterState clusterState) {
             // Index must have been since deleted, ignore it
             return clusterState;
         }
+
         LifecycleExecutionState lifecycleState = LifecycleExecutionState
             .fromIndexMetadata(indexMetaData);
+
+        if (shouldParseIndexName(indexMetaData.getSettings())) {
```

Comment:
I think this can be refactored a bit not to have multiple return paths for the cluster state, instead using a single `ClusterState.Builder` and only adding updated settings iff `shouldParseIndexName` is true